pkg/syncer/statussyncer.go

```diff
@@ -64,7 +70,60 @@ func NewStatusSyncer(from, to *rest.Config, gvrs []string, kcpClusterName logica
 	return New(kcpClusterName, pclusterID, fromClient, toClient, SyncUp, gvrs, pclusterID, mutatorsMap)
 }
 
-func (c *Controller) updateStatusInUpstream(ctx context.Context, gvr schema.GroupVersionResource, upstreamNamespace string, downstreamObj *unstructured.Unstructured) error {
+func (c *Controller) deleteFromUpstream(ctx context.Context, gvr schema.GroupVersionResource, upstreamNamespace, name string) error {
+	return c.removeFinalizersAndUpdate(ctx, c.toClient, gvr, upstreamNamespace, name)
+}
+
+func (c *Controller) removeFinalizersAndUpdate(ctx context.Context, upstreamClient dynamic.Interface, gvr schema.GroupVersionResource, upstreamNamespace, name string) error {
```

Comment:
give it a better name. Is this about upstream? Downstream?

Reply:
Done